src/plugins/data_source_management/public/components/direct_query_data_sources_components/acceleration_creation/selectors/source_selector.tsx

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiDescriptionList,
+  EuiDescriptionListDescription,
+  EuiDescriptionListTitle,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import producer from 'immer';
+import React, { useEffect, useState } from 'react';
+import { HttpStart, NotificationsStart } from 'opensearch-dashboards/public';
+import { DATACONNECTIONS_BASE } from '../../../../../framework/utils/shared';
+import {
+  CachedDataSourceStatus,
+  CachedDatabase,
+  CreateAccelerationForm,
+} from '../../../../../framework/types';
+import { CatalogCacheManager } from '../../../../../framework/catalog_cache/cache_manager';
+import { hasError, validateDataTable, validateDatabase } from '../create/utils';
+import { SelectorLoadDatabases } from './selector_helpers/load_databases';
+import { SelectorLoadObjects } from './selector_helpers/load_objects';
+
+interface AccelerationDataSourceSelectorProps {
+  http: HttpStart;
+  notifications: NotificationsStart;
+  accelerationFormData: CreateAccelerationForm;
+  setAccelerationFormData: React.Dispatch<React.SetStateAction<CreateAccelerationForm>>;
+  selectedDatasource: string;
+  dataSourcesPreselected: boolean;
+  tableFieldsLoading: boolean;
+  dataSourceMDSId?: string;
+}
+
+export const AccelerationDataSourceSelector = ({
+  http,
+  notifications,
+  accelerationFormData,
+  setAccelerationFormData,
+  selectedDatasource,
+  dataSourcesPreselected,
+  tableFieldsLoading,
+  dataSourceMDSId,
+}: AccelerationDataSourceSelectorProps) => {
+  const [databases, setDatabases] = useState<Array<EuiComboBoxOptionOption<string>>>([]);
+  const [selectedDatabase, setSelectedDatabase] = useState<Array<EuiComboBoxOptionOption<string>>>(
+    []
+  );
+  const [tables, setTables] = useState<Array<EuiComboBoxOptionOption<string>>>([]);
+  const [selectedTable, setSelectedTable] = useState<Array<EuiComboBoxOptionOption<string>>>([]);
+  const [loadingComboBoxes, setLoadingComboBoxes] = useState({
+    dataSource: false,
+    database: false,
+    dataTable: false,
+  });
+
+  const dataSourceDescription = (
+    <EuiDescriptionList>
+      <EuiDescriptionListTitle>Data source</EuiDescriptionListTitle>
+      <EuiDescriptionListDescription>
+        {accelerationFormData.dataSource}
+      </EuiDescriptionListDescription>
+    </EuiDescriptionList>
+  );
+
+  const loadDataSource = () => {
+    setLoadingComboBoxes({ ...loadingComboBoxes, dataSource: true });
+    http
+      .get(`${DATACONNECTIONS_BASE}/dataSourceMDSId=${dataSourceMDSId ?? ''}`)
+      .then((res) => {
+        const isValidDataSource = res.some(
+          (connection: any) =>
+            connection.connector.toUpperCase() === 'S3GLUE' &&
```

Comment:
nit. I feel we want to align our way to do these checks, there's checking in observability at least where it converts uppercase to lowercase then compare but here it converts back to uppercase and comparing it with a hardcoded string, there's constant out there with value S3GLUE exposed, if we want to convert to uppercase then reuse the constant or don't do conversion and compare directly with a lowercase

Reply:
I agree. This will be added into the followup.